distributed/worker.py

```diff
@@ -2759,18 +2767,37 @@ def get_client(address=None, timeout=None, resolve_address=True) -> Client:
         pass
     else:
         if not address or worker.scheduler.address == address:
-            return worker._get_client(timeout=timeout)
-
-    from distributed.client import Client
+            client = worker._get_client(timeout=timeout)
+            if asynchronous is None or client._asynchronous == asynchronous:
+                return client
+            return Client(
```

Comment:
it's not clear who's responsibility it is to `async with` this Client, eg only newly created clients need it. I think a new api should be introduced that's always a context manager rather than adding a flag to `get_client`